isis/src/control/objs/BundleConstraint/BundleLidarRangeConstraint.cpp

```diff
@@ -0,0 +1,440 @@
+#include "BundleLidarRangeConstraint.h"
+
+// Qt Library
+#include <QDebug>
+
+// Boost uBLAS
+#include <boost/numeric/ublas/vector_proxy.hpp>
+
+// Isis Library
+#include "Camera.h"
+#include "CameraGroundMap.h"
+#include "IsisBundleObservation.h"
+#include "SpicePosition.h"
+
+using namespace boost::numeric::ublas;
+
+namespace Isis {
+
+  /**
+   * constructor
+   *
+   * @param lidarControlPoint input lidar control point
+   * @param measure associated BundleMeasureQsp.
+   */
+  BundleLidarRangeConstraint::BundleLidarRangeConstraint(LidarControlPointQsp lidarControlPoint,
+                                                         BundleMeasureQsp measure) {
+    m_lidarControlPoint   = lidarControlPoint;
+    m_simultaneousMeasure = measure;
+    m_bundleObservation   = measure->parentBundleObservation();
+    m_rangeObserved       = m_lidarControlPoint->range();
+
+    if (m_rangeObserved <= 0) {
+      QString msg ="In BundleLidarRangeConstraint::BundleLidarRangeConstraint():"
+                   "observed range for lidar point must be positive (Point Id: "
+                   + measure->parentControlPoint()->id() + ")\n.";
+      throw IException(IException::Programmer, msg, _FILEINFO_);
+    }
+
+    m_rangeObservedSigma = m_lidarControlPoint->sigmaRange() * 0.001; // converting from m to km
+
+    if (m_rangeObservedSigma <= 0) {
+      QString msg ="In BundleLidarRangeConstraint::BundleLidarRangeConstraint():"
+                   "observed range sigma for lidar point must be positive (Point Id: "
+                   + measure->parentControlPoint()->id() + ")\n.";
+      throw IException(IException::Programmer, msg, _FILEINFO_);
+    }
+
+    m_rangeObservedWeightSqrt = 1.0/m_rangeObservedSigma;
+    m_adjustedSigma = 0.0;
+    m_dX = m_dY = m_dZ = 0.0;
+
+    // Check that the simultaneous image has an ISIS camera
+    if (m_simultaneousMeasure->camera()->GetCameraType() == Camera::Csm) {
+      QString msg = "Cannot apply a Lidar range constraint to a CSM camera model"
+                    "(Point Id: "
+                    + measure->parentControlPoint()->id() + ", Measure Serial:"
+                    + measure->cubeSerialNumber() + ").\n";
+      throw IException(IException::Programmer, msg, _FILEINFO_);
+    }
```

Comment:
Should a check like this occur every time a camera is called in this file?

Reply:
No, that would be redundant because you can't call any of the other functions without calling this constructor first.